packages/web/src/components/buy-sell-modal/hooks/useTokenBalanceManager.ts

```diff
@@ -0,0 +1,123 @@
+import { useMemo } from 'react'
+
+import {
+  useAudioBalance,
+  useTokenBalance,
+  getTokenMintName
+} from '@audius/common/api'
+import { Status } from '@audius/common/models'
+import { TokenInfo } from '@audius/common/store'
+import { isNullOrUndefined } from '@audius/common/utils'
+import { AUDIO, FixedDecimal } from '@audius/fixed-decimal'
+
+import type { BalanceConfig } from '../types'
+
+export type TokenBalanceManager = {
+  inputBalance: BalanceConfig
+  outputBalance: BalanceConfig
+}
+
+/**
+ * Shared hook for managing token balance fetching and formatting
+ * Used by both ConvertTab and SellTab to eliminate code duplication
+ */
+export const useTokenBalanceManager = (
+  inputToken: TokenInfo,
+  outputToken: TokenInfo
+): TokenBalanceManager => {
+  // Fetch AUDIO balance for current user
+  const { accountBalance } = useAudioBalance({ includeConnectedWallets: false })
```

Comment:
can we update useAudioBalance to give back status? makes the status variables below easier to compute